core/src/main/resources/swagger/api-docs.yaml

```diff
@@ -4411,6 +4411,22 @@ components:
           type: string
           description: A description for the snapshot
           default: ""
+        properties:
+          type: object
+          description: A map of key-value pairs to associate with the snapshot. Optional, defaults to an empty map if not specified.
+          default: {}
+        cloningInstructions:
+          type: string
+          default: COPY_NOTHING
+          description: |
+            Cloning semantics for this snapshot. Optional, defaults to COPY_NOTHING if not specified.
+            COPY_NOTHING: Don't clone the snapshot.
+            COPY_REFERENCE: Create a new referenced snapshot that points to same cloud resource as the source snapshot.
+            COPY_LINK_REFERENCE: Create a new referenced snapshot that points to the same cloud resource as the source resource, AND link the source workspace policy to the destination workspace policy; changes in the source will propagate to the destination.
```

Comment:
does this infer that `COPY_REFERENCE` does NOT do anything with policy?

I have workspace A.
I link snapshot 1 to workspace A. This links snapshot 1's policies to workspace A.
I clone workspace A to workspace B. After cloning:

if the A:1 link is `COPY_REFERENCE`:
* workspace B contains a reference to snapshot 1
* workspace B does _not_ have snapshot 1's policies

if the A:1 link is `COPY_LINK_REFERENCE `:
* workspace B contains a reference to snapshot 1
* workspace B _does_ have snapshot 1's policies

I'm unclear on why this would be different. When I add the reference from workspace A to snapshot 1, the snapshot policies get copied onto the workspace. At that point, they're the workspace's policies; even if I remove the reference, the policies remain on workspace A.

Wouldn't cloning workspace A copy all of its policies, regardless of where they came from, to workspace B?

Reply:
Full disclosure, this is [pulled](https://workspace.dsde-dev.broadinstitute.org/swagger-ui.html#/ReferencedGcpResource/createDataRepoSnapshotReference) from the WSM API docs and is a straight passthrough. 

The distinction here is whether the policies are linked or merged. The COPY_REFERENCE semantics merge the policies on clone, whereas the LINK_REFERENCE _links_ the policies (the TPS [design doc](https://docs.google.com/document/d/1dRtE_nZf8e231DZmk_-9CDbjYTiIN2ZZwI95uF1w2pg/edit#heading=h.1uc5ubqdd419) gets into the details). Here is a useful snippet: 

```
TPS supports merging the policies from one policy object to another without creating a dependency between the source of the merge and the target. We expect this to be used to model the policy effects when workspaces or resources in workspaces are copied (cloned).

The idea is that when a merge is done, the destination PAO must enforce all of the policies that are on the originating PAO. However, later on new policies might be added to the originating PAO. Those should be of no consequence to the destination PAO.
```
In both examples above, the destination workspace should receive snapshot 1's policies. My understanding from the doc and the above snippet is that merging is present to prevent the policy modifications from the source PAO propagating to the cloned one. 

If this is confusing, maybe we limit this enum to solely NOTHING or COPY_REFERENCE?

Reply:
got it, thank you, I appreciate the explanation! I understand that `COPY_REFERENCE` is suggested for cloning, so baseline we'll go with that. 

I also feel like `COPY_LINK_REFERENCE` is what we *should* suggest for cloning, but that's a business decision and I will check on that internally with Product; no changes requested for this PR.